### ISSUE TITLE
Mirror hangs and abort isn't possible

### ISSUE THREAD

**Issue by somera** [go-gitea/gitea#29349]
### Description

Just try to mirror this

https://github.com/testcontainers/testcontainers-dotnet-legacy.git

project.

It isn't wotking with 1.21.6 and on https://try.gitea.io/ .

![image](https://github.com/go-gitea/gitea/assets/8334250/d2856ef8-bcf5-40c5-93f7-624549105b77)

There is no error in log. And you can't abbort it. Only possible way is to kill the cron task in the admin view (/admin/monitor/stacktrace?show=process).

Just run it and you can reproduce it.

### Gitea Version

1.21.6

### Can you reproduce the bug on the Gitea demo site?

Yes

### Git Version

2.43.2

### Operating System

Ubuntu 22.04.4 LTS x86_64

### How are you running Gitea?

Downloaded gitea-1.21.6-linux-amd64

### Database

PostgreSQL

**Comment by somera** [comment]
Still running on https://try.gitea.io/ .

![image](https://github.com/go-gitea/gitea/assets/8334250/d2bedc9e-d4c3-4007-8440-c2195832f5d5)

**Comment by yp05327** [comment]
I successfully mirrored it:
https://try.gitea.io/yp05327/testcontainers-dotnet-legacy

my options:
![image](https://github.com/go-gitea/gitea/assets/18380374/854bada8-1c95-41f5-ab84-e25c2e87b40a)

**Comment by yp05327** [comment]
I can reproduce it in 1.21.6 in windows.
![image](https://github.com/go-gitea/gitea/assets/18380374/3398d678-3ebc-42f7-ab29-561e732999e5)
The first error is mirror error, the rest of them happened when I click cancel button.

Then in 1.21.7, I can get same error but it will be mirrored successfully:
![image](https://github.com/go-gitea/gitea/assets/18380374/80eab67b-24d4-4b62-9fb8-4553740b8e6f)

**Comment by yp05327** [comment]
The mirror is success, when access the home page, it froze in reading `.editorconfig` in `GetEditorconfig`
Maybe it is related to #29301. If I revert it, it will not froze in reading `.editorconfig`, but in `editorconfig.ParseGraceful`. 😕

**Comment by wxiaoguang** [comment]
> The mirror is success, when access the home page, it froze in reading `.editorconfig` in `GetEditorconfig` Maybe it is related to #29301. If I revert it, it will not froze in reading `.editorconfig`, but in `editorconfig.ParseGraceful`. 😕

The "frozen" bug should has been fixed by 1.21.7

> Then in 1.21.7, I can get same error but it will be mirrored successfully:

It could be another bug (maybe a longstanding problem, not related to this issue)


----

To be clear: 1.21.7 on windows still hangs, right?

**Comment by yp05327** [comment]
> The "frozen" bug should has been fixed by 1.21.7

Ah, I tested in 1.21.6, I will try it in 1.21.7

**Comment by yp05327** [comment]
Tried it again in 1.21.7, I got the error on windows (both git and gogit version), but not on ubuntu
``` text
...it/repo_attribute.go:315:func2() [E] Unable to open checker for b4cc5165369ba74d667d91dcc0b637a09bd05f74. Error: failed to run attr-check. Error: exit status 1
Stderr:
```
git version on windows is `2.40.0.windows.1`

As this is another issue, I will create a new one.

@somera 
I will close this issue, as it worked after upgrade to 1.21.7
Feel free to reopen it if you have any questions. 😄
